### DISCUSSION TITLE
Provide the bitcoin address in Greenfield V1 Invoice API

### DISCUSSION THREAD

**Discussion by isaacy** [btcpayserver/btcpayserver#2291] [Technical Support]
The BitPay API is able to return the Bitcoin address and other exchange rates information.  Would like to request Greenfield API to support return the same info, or at least, the Bitcoin address.  It is critical for businesses/merchant that would want to create their own checkout page and generate their own QR code

**Comment by isaacy** [comment]
The ask is about adding the following to the Greenfield parameters

btcPrice: '0.00019471',
  btcDue: '0.00019471',
  cryptoInfo: [
    {
      cryptoCode: 'BTC',
      paymentType: 'BTCLike',
      rate: 51358.729,
      exRates: [Object],
      paid: '0.00000000',
      price: '0.00019471',
      due: '0.00019471',
      paymentUrls: [Object],
      address: '<btc_address>',
      url: <URL>,
      totalDue: '0.00019471',
      networkFee: '0.00000000',
      txCount: 0,
      cryptoPaid: '0.00000000',
      payments: []
    }
  ],

addresses: { BTC: <btc_address> },
  paymentCodes: {
    BTC: {
      BIP21: '<btc_address?amount=0.00019471',
      BIP72: null,
      BIP72b: null,
      BIP73: null,
      BOLT11: null
    }
  },

**Reply by pavlenex** [reply]
https://docs.btcpayserver.org/API/Greenfield/v1/#operation/StoreOnChainWallets_GetOnChainWalletReceiveAddress ?

**Reply by mp21-org** [reply]
this would probably be better suited to solve your issue:
https://docs.btcpayserver.org/API/Greenfield/v1/#operation/Invoices_GetInvoicePaymentMethods

check the destination field of the returned data array.
